devito/grid.py

```diff
@@ -107,11 +116,51 @@ def __repr__(self):
             self.extent, self.shape, self.dimensions
         )
 
+    @property
+    def extent(self):
+        """Physical extent of the domain in m."""
+        return self._extent
+
+    @property
+    def dtype(self):
+        """Data type inherited by all :class:`Function`s defined on this Grid."""
+        return self._dtype
+
+    @property
+    def origin(self):
+        """Physical coordinates of the domain origin."""
+        return self._origin
+
+    @property
+    def dimensions(self):
+        """Spatial dimensions of the computational domain."""
+        return self._dimensions
+
     @property
     def dim(self):
         """Problem dimension, or number of spatial dimensions."""
         return len(self.shape)
 
+    @property
+    def time_dim(self):
+        """Time dimension associated with this Grid."""
+        return self._time_dim
+
+    @property
+    def stepping_dim(self):
+        """Stepping dimension associated with this Grid."""
+        return self._stepping_dim
+
+    @property
+    def subdomains(self):
+        """The :class:`SubDomain`s defined in this Grid."""
+        return {i.name: i for i in self._subdomains}
+
+    @property
+    def interior(self):
```

Comment:
This is kind of the same as @mloubout 's comment further down about how thick the default should be.  I'd be quite surprised if anything other than the tests used Eq(region=INTERIOR) in its current form - it doesn't work for >2nd order spatial stencils, and you don't need it if you only want zeros in the boundary anyway.  So that doesn't leave a lot of useful scenarios.

Since this PR is (rightfully) breaking compatibility by changing the syntax for `Eq` away from `region=INTERIOR`, should we take the opportunity to retire it completely, or change it to something more generic rather than carry it around?  And the relevant tests could create a subdomain themselves, rather than using this property?